provider/blockdev_base.py

```diff
@@ -95,7 +95,11 @@ def generate_data_file(self, tag, filename=None):
         timeout = params.get_numeric("create_tempfile_timeout", 720)
         backup_utils.generate_tempfile(
             self.main_vm, self.disks_info[tag][1], filename, image_size, timeout)
-        self.files_info[tag] = [filename]
+
+        if tag not in self.files_info:
+            self.files_info[tag] = [filename]
+        else:
+            self.files_info[tag].append(filename)
```

Comment:
And this file , please add commit for it to declare the reason why to modify it.

Reply:
split into separate commits for provider